src/lib/cloud-sdk-api.js

```diff
@@ -191,6 +245,141 @@ class CloudSdkAPI {
+      }
+
+      return status;
+  }
+
+  async _waitForEnvRunning(namespace) {
+    await this._waitForEnv(namespace, 'running');
+  }
+
+  async _waitForEnvHibernated(namespace) {
+    await this._waitForEnv(namespace, 'hibernated');
+  }
+
+  async _waitForEnvRunningOrHibernated(namespace) {
+    return await this._waitForEnv(namespace, 'running', 'hibernated');
+  }
+
+  async _waitForEnv(namespace, state1, state2) {
+    return (await this._waitForStatus(
+      (status) => status.releases?.status[`cm-p${this._programId}-e${this._environmentId}`]?.releaseState === state1 ||
+                  status.releases?.status[`cm-p${this._programId}-e${this._environmentId}`]?.releaseState === state2,
+      async () => await this._doDevConsoleRequest(`get`,`/api/releases/${namespace}/status`)
+    )).releases.status[`cm-p${this._programId}-e${this._environmentId}`].releaseState;
+  }
+
+  async _hibernateEnv(namespace) {
+    await this._setEnvStatus(namespace, `hibernate`);
+  }
+
+  async _dehibernateEnv(namespace) {
+    await this._setEnvStatus(namespace, `dehibernate`);
+  }
+
+  async _setEnvStatus(namespace, status) {
+    await this._waitForStatus(
+      (status) => status.ok,
+      async () => await this._doDevConsoleRequest(`post`, `/api/releases/${namespace}/${status}/cm-p${this._programId}-e${this._environmentId}`)
+    );
+  }
+
+  async _getNamespace() {
+    const nameSpaceRequest = await this._doDevConsoleRequest(`get`, `/api/status`);
+    if (nameSpaceRequest.status === 200) {
+      const nameSpaceStatus = await nameSpaceRequest.json();
+      if (nameSpaceStatus.availableNamespaces && nameSpaceStatus.availableNamespaces[0]) {
+        return nameSpaceStatus.availableNamespaces[0];
+      } else {
+        throw `Error: no namespace found`;
+      }
+    } else {
+      throw `Error: ${nameSpaceRequest.status} - ${nameSpaceRequest.statusText}`;
+    }
+  }
+
+  async _checkRDE() {
+    const response = await this.getArtifacts(`limit=0`)
+    if (response.status !== 200) {
+        throw `Error: ${response.status} - ${response.statusText}`
+    }
+  }
+
+  async _startEnv(namespace) {
+    await this._dehibernateEnv(namespace);
+    await this._waitForEnvRunning(namespace);
+  }
+
+  async _stopEnv(namespace) {
+    await this._hibernateEnv(namespace);
+    await this._waitForEnvHibernated(namespace);
+  }
+
+  async startEnv() {
+    await this._checkRDE();
+    const namespace = await this._getNamespace();
+    let status = await this._waitForEnvRunningOrHibernated(namespace);
+    if (status == 'hibernated') {
+      await this._startEnv(namespace);
+    } else {
+      throw `Error: environment not hibernated`
+    }
+  }
+
+  async stopEnv() {
+    await this._checkRDE();
+    const namespace= await this._getNamespace();
+    let status = await this._waitForEnvRunningOrHibernated(namespace);
+    if (status == 'running') {
+      await this._stopEnv(namespace);
+    } else {
+      throw `Error: environment not running`
```

Comment:
Is it an error if the env is already stopped/hibernated?

Reply:
The idea was that for start and stop it is an error. If you want something else, you have to use _start/_stop (which doesn't check).